docs/Conformance-Guide-draft.md

```diff
@@ -0,0 +1,103 @@
+
+1. (Required) Support: **The extension needs to document how a user would get support or file issues for the extension for problems that are not related to Zowe and Zowe Explorer.**
+
+   The extender needs to document where the user can find this information in the conformance application.
+
+1. (Best Practice) User settings consistency: **For a consistent user experience we recommend that user settings and configuration settings follow the naming conventions as documented in the Zowe Explorer extensibility documentation.**
+
+   The list and documentation of the configuration settings should be made available as part of the compliance application. If the extension is a VS Code extensions then including the package.json file will be sufficient unless other means of configuration settings (such as environment variables) have been used.
+
+1. (Best Practice) Error message consistency: **For a consistent user experience error and log messages should follow the format documented in the Zowe Explorer extensibility documentation.**
+
+   To verify this requirement it is sufficient to provide links to user documentation showing screen shots of error dialogs as well as sample log file entries.
+
+1. (Best Practice) Zowe SDK usage: **Utilize available Zowe SDKs to standardize on z/OS interactions as well as other common capabilities used by many other Zowe extensions and tools.**
+
+   This can be verified by reviewing the dependencies list of the implementation such as in the package.json file of a nodejs-based extension.
+
+## Profiles Access extension conformance criteria (1)
+
+A Profiles Access extension is a Zowe Explorer extension that uses the Zowe Extensibility API to conveniently access Zowe CLI profiles loaded by Zowe Explorer itself. This allows the extension to consistently access profile instances of specific types, offer them for edit and updates as well as common refresh operations that apply to all extensions, add more profile types it is using itself for its own custom views (for example a CICS extension adding a CICS explorer view) and other similar use cases related to Zowe CLI profiles. These extensions do **not** have to be VS Code extension if it just wants to use ProfilesCache implementation of Zowe Explorer as all APIs are provided free of any VS Code dependencies. Such an extension could be used for another non-VS Code tool, a Zowe CLI plugin, a Web Server or another technology. However, to access the profiles cache of the actual running VS Code Zowe Explorer the extender needs to be a VS Code extension that has an extension dependency defined to be able to query the extender APIs. Therefore, some of the criteria that are listed here as required are only required if the extender is a VS Code extension.
+
```

Comment:
Additional entries:
- (Required) **Sharing of profiles with Zowe CLI:** Extensions shares the created profiles between Zowe CLI and VSCode
- (V1: Best Practice, V2: Required) Extension supports base profiles and tokens
- (V1: Best Practice, V2: Required) Extension supports JSON configuration file
- (V1: Best Practice, V2: Required) If the extension supports Zowe's Secure Credential store, it calls the Zowe Explorer-provided method

Reply:
Another one before moving to dependencies:

The extension is built on top of at least one Zowe Explorer provided API.

[Select all that apply]
a. Dependency (Extension Accessing Profiles)
b. Dependency (Data Provider Extension)
c. Dependency (Extension Adding Menus)